### PR TITLE
Hw jv web practice

### PR DESCRIPTION
**Pull request by vika-trach**

### CODE REVIEW

File: src/main/java/mate/controller/CreateDriverController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Injector injector = Injector.getInstance("mate");
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Driver driver = new Driver();
+        driver.setName(req.getParameter("name"));
+        driver.setLicenseNumber(req.getParameter("license_number"));
```
Comment (Gamemod13) [inline src/main/java/mate/controller/CreateDriverController.java:23]: Better use constructor to initialization driver

File: src/main/java/mate/controller/CreateManufacturerController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    private final ManufacturerService manufacturerService
+            = (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setName(req.getParameter("name"));
+        manufacturer.setCountry(req.getParameter("country"));
+        manufacturerService.create(manufacturer);
```
Comment (Gamemod13) [inline src/main/java/mate/controller/CreateManufacturerController.java:24]: Better use constructor to initialization

File: src/main/java/mate/controller/IndexController.java [hunk 3]
```diff
@@ -2,10 +2,12 @@
...
 import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+@WebServlet(urlPatterns = "/index")
 public class IndexController extends HttpServlet {
     @Override
     public void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (Gamemod13) [inline src/main/java/mate/controller/IndexController.java:13]: Please, use one type of access modifier for all servlets for consistency. In all previous classes you used protected, so it's will be better use it here too 
```suggestion
    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```

File: src/main/java/mate/model/Car.java [hunk 4]
```diff
@@ -9,6 +9,16 @@ public class Car {
...
     private List<Driver> drivers;
 
+    public Car(String model, Manufacturer manufacturer, List<Driver> drivers) {
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = drivers;
+    }
+
+    public Car() {
+
```
Comment (Gamemod13) [inline src/main/java/mate/model/Car.java:19]:
```suggestion
```

File: src/main/java/mate/model/Driver.java [hunk 5]
```diff
@@ -7,6 +7,15 @@ public class Driver {
     private String name;
     private String licenseNumber;
 
+    public Driver(String name, String licenseNumber) {
+        this.name = name;
+        this.licenseNumber = licenseNumber;
+    }
+
+    public Driver() {
+
```
Comment (Gamemod13) [inline src/main/java/mate/model/Driver.java:16]:
```suggestion
```

File: src/main/java/mate/model/Manufacturer.java [hunk 6]
```diff
@@ -7,6 +7,15 @@ public class Manufacturer {
     private String name;
     private String country;
 
+    public Manufacturer(String name, String country) {
+        this.name = name;
+        this.country = country;
+    }
+
+    public Manufacturer() {
+
```
Comment (Gamemod13) [inline src/main/java/mate/model/Manufacturer.java:16]:
```suggestion
```

**Review by Gamemod13 — CHANGES_REQUESTED** [review]
Very good work. You can easy make it better after this minor fixes

File: src/main/java/mate/model/Car.java (hunk 4 above)
Comment (vika-trach) [reply src/main/java/mate/model/Car.java:19]: It's maven rule to have one empty line between methods

**Review by Gamemod13 — APPROVED** [review]
Good! Keep pushing)

File: src/main/java/mate/controller/GetAllManufacturersController.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.setAttribute("manufacturers", manufacturerService.getAll());
+        req.getRequestDispatcher("/WEB-INF/views/manufacturers/all.jsp")
+                .forward(req, resp);
+    }
+}
+
```
Comment (sarakhmen) [inline src/main/java/mate/controller/GetAllManufacturersController.java:26]:
```suggestion
```

File: src/main/java/mate/model/Car.java (hunk 4 above)
Comment (sarakhmen) [reply src/main/java/mate/model/Car.java:19]: probably @Gamemod13 meant an empty line within a constructor

**Review by sarakhmen — APPROVED** [review]
good job ;)

**Review by Nick97-git — APPROVED** [review]
test

**Pull request closed without merging**